File: src/test/java/eotree/data/TestEOStringData.java
package eotree.data;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;
import org.polystat.j2eo.eotree.data.EOStringData;

/**
 * EO string data tests.
 */
public class TestEOStringData {

    @Test
    public void testGenerateEOZeroIndent() {
        var s = new EOStringData("hello world");

        assertEquals(s.generateEO(0), "\"hello world\"");
    }

    @Test
    public void testGenerateEONonZeroIndent() {
        var f = new EOStringData("hello world");

        assertEquals(f.generateEO(1), "  \"hello world\"");
    }
}

File: src/test/kotlin/eotree/data/TestEOStringData.kt
package eotree.data

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.polystat.j2eo.eotree.data.EOStringData

/**
 * EO string data tests.
 */
class TestEOStringData {
    @Test
    fun testGenerateEOZeroIndent() {
        val s = EOStringData("hello world")
        Assertions.assertEquals(s.generateEO(0), "\"hello world\"")
    }

    @Test
    fun testGenerateEONonZeroIndent() {
        val f = EOStringData("hello world")
        Assertions.assertEquals(f.generateEO(1), "  \"hello world\"")
    }
}
